libnavui-androidauto/src/main/java/com/mapbox/androidauto/navigation/MapboxCarNavigationManager.kt

```diff
@@ -42,19 +43,26 @@ class MapboxCarNavigationManager internal constructor(
         onUpdateTrip(trip)
     }
 
-    private val tripSessionStateObserver = TripSessionStateObserver { tripSessionState ->
-        logAndroidAuto("$LOG_CATEGORY tripSessionStateObserver state: $tripSessionState")
-        when (tripSessionState) {
-            TripSessionState.STARTED -> navigationManager.navigationStarted()
-            TripSessionState.STOPPED -> navigationManager.navigationEnded()
+    private var inActiveNavigation = false
+
+    private val routesObserver = RoutesObserver {
+        if (it.navigationRoutes.isEmpty()) {
+            logAndroidAuto("$LOG_CATEGORY stop active navigation")
+            inActiveNavigation = false
+            navigationManager.navigationEnded()
```

Comment:
Do you know if calling `navigationEnded` adds any limitations that might make the app not functional in free drive mode?

Reply:
I haven't found any, location updates work in FreeDrive after `navigationEnded`.